### ISSUE TITLE
ayo this is mediocre coding

### ISSUE THREAD

**Issue by rbhog** [aidangoettsch/twitch-ocr#2]
I was entranced by the idea of an automatic script running Python™ code to automatically look at the comments being posted by individuals on the streaming platform, Twitch™. As I looked through your code I was astonished for the most part at the immaculateness of this code, however, I was struck with sudden shock as I saw the following line of code:
`streams = streamlink.streams('https://www.twitch.tv/%s' % streamer)`

Your code is unoptimized as shown by the following disassembly comparison of an f string and the % operator for string interpolation.

With the following code for reference:
```
import dis


def f_string(thing: str):
    return f'{thing}'


def format_string(thing: str):
    return "%s" % thing



if __name__ == '__main__':
    print("============================== f-strings ==============================")
    dis.dis(f_string)
    print("============================== format thingy ==========================")
    dis.dis(format_string)
```

I was able to get the following output:

```
============================== f-strings ==============================
 15           0 LOAD_FAST                0 (thing)
              2 FORMAT_VALUE             0
              4 RETURN_VALUE
============================== format thingy ==========================
 19           0 LOAD_CONST               1 ('%s')
              2 LOAD_FAST                0 (thing)
              4 BINARY_MODULO
              6 RETURN_VALUE
```

My results lead me to believe that your code has suboptimal performance that could be enhanced in order to maximize my Twitch™ live chat viewing experience.

**Comment by cf12** [comment]
i agree with this sentiment

**Comment by rbhog** [comment]
I would like to add that I have more findings to present. Using the following code to analyze the performance of the mod operator, the f-string, and the format method:

```
import timeit
import matplotlib.pyplot as plt

times_mod = []
times_fstring = []
times_format = []
x_ac = range(50)

test = """
abc
"""

test = """
i = "discord"
'https://www.twitch.tv/%s' % i
"""

for i in range(50):
    times_mod.append(timeit.timeit(stmt=test))

test = """
i = "discord"
'https://www.twitch.tv/{}'.format(i)
"""

for i in range(50):
    times_format.append(timeit.timeit(stmt=test))

test = """
i = "discord"
f'https://www.twitch.tv/{i}'
"""

for i in range(50):
    times_fstring.append(timeit.timeit(stmt=test))

g1 = (x_ac, times_mod)

g2 = (x_ac, times_format)

g3 = (x_ac, times_fstring)

data = (g1, g2, g3)
colors = ("red", "green", "blue")
groups = ("mod", "format", "fstring")

fig = plt.figure()
ax = fig.add_subplot(1, 1, 1)

for data, color, group in zip(data, colors, groups):
    x, y = data
    ax.scatter(x, y, alpha=0.8, c=color, edgecolors='none', s=30, label=group)

plt.title('Comparison')
plt.legend(loc=2)
plt.show()
```

I was able to produce the following graph, detailing the time spent per option.


![](https://cdn.discordapp.com/attachments/416776042733109260/842448986405077043/unknown.png)

This leads me to believe that the f-string is indeed the most optimal string type for everyone who uses this code and my  Twitch™ live chat viewing experience.

**Comment by aidangoettsch** [comment]
Fixed by #3

**Comment by rbhog** [comment]
As a final update, I wanted to calculate the standard deviation of each of the datasets. Given the following graph representing the runtimes for each of the functions in question:

![](https://cdn.discordapp.com/attachments/416776042733109260/842454397539582022/unknown.png)


I was able to get the following standard deviations:

```
============================== f-strings ==============================
0.001822783955332215
=============================== mod opr ===============================
0.004349262570572507
=============================== formats ===============================
0.005637659276881349
```
